ROADMAP.md

```diff
@@ -0,0 +1,57 @@
+# Roadmap
+
+The OpenAssetIO project is working towards an initial beta release of
+the core API.
+
+We are looking for early adopters who can help fine tune its final form,
+and validate key design choices, and help flesh out asset types and
+their properties.
+
+The project is functional, and moving forward we expect minimal churn in
+the API itself as we complete the implementation of language specific
+features.
+
+## Y23 Q3: v1.0.0b1
+
+This release will support resolution and publishing of entities in batch
+and interactive context, along with relationship discovery.
+
+**Hosts:** C++, Python
+**Managers:** Python
+
+- [x] Migration to the [Trait based API](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/doc/decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md).
+- [x] Migration to batch-centric API.
+- [-] Migration of core API methods to C+.
+- [ ] Sundry minor API Change issues.
+
+## Y24 Q1: v1.0.0
+
+This release will provide additional introspection methods and
+debugging functionality, along with language parity between Hosts and
+Managers.
+
+**Hosts:** C++, Python
+**Managers:** C++, Python
+
+- [ ] Auth related error codes.
+- [ ] Debug trace logging support.
+- [ ] Entity introspection API methods.
+- [ ] C++ Plugin System
+- [ ] Hybrid C++/Python manager bridge.
+- [ ] Out-of-process Python
+- [ ] Migrate landing page/examples to [OpenAssetIO-MediaCreation](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation)
+
+## Y24 H1: OpenAssetIO UI v1.0.0a1
```

Comment:
[consider] 

- #7

Reply:
This was another we were waiting for real-world use cases for before it was prioritised as its incredibly hypothetical at present.

Reply:
We could take a look at what few limited integrations currently exist and we have visibility of the code (Ayon, Shotgrid) to see what, if any, SDK access they require beyond what OpenAssetIO can/will provide, and get an idea how much priority they should be given. 

Also ASWF/studio contacts could be asked (though I suspect limited help there).

Or maybe its a case of we accept that SDK access will be required, but that's not the responsibility of OpenAssetIO, and at best OpenAssetIO will help inform the context (node name, etc) to make bespoke SDK usage a little bit easier.

Reply:
Yeah 100% Just trying to balance hours in the day and functionality that is currently blocking in-application workflows... trying not to have too many things in flight at once.